Patches/Geth/Ammo_Geth_CE.xml

```diff
@@ -0,0 +1,414 @@
+<Patch>
+
+	<Operation Class="PatchOperationFindMod">
+    <mods>
+		<li>Geth</li>
+    </mods>
+		<match Class="PatchOperationSequence">
+		<operations>		
+
+		<li Class="PatchOperationAdd">
+		<xpath>Defs</xpath>
+		<value>
+
+  <ThingCategoryDef>
+    <defName>AmmoML77CE</defName>
+    <label>ML-77 Missile</label>
+    <parent>AmmoRockets</parent>
+    <iconPath>UI/Icons/ThingCategories/CaliberRocket</iconPath>
+  </ThingCategoryDef>
+
+	<!-- ==================== AmmoSet ========================== -->
+
+	<CombatExtended.AmmoSetDef>
+		<defName>AmmoSet_GethRifle</defName>
+		<label>Geth thermal clip</label>
+		<ammoTypes>
+			<Ammo_Geth>Bullet_GethRifle</Ammo_Geth>
+		</ammoTypes>
+	</CombatExtended.AmmoSetDef>
+
+	<CombatExtended.AmmoSetDef>
+		<defName>AmmoSet_GethSMG</defName>
+		<label>Geth thermal clip</label>
+		<ammoTypes>
+			<Ammo_Geth>Bullet_GethSMG</Ammo_Geth>
+		</ammoTypes>
+	</CombatExtended.AmmoSetDef>
+
+
+	<CombatExtended.AmmoSetDef>
+		<defName>AmmoSet_GethShotgun</defName>
+		<label>Geth thermal clip</label>
+		<ammoTypes>
+			<Ammo_Geth>Bullet_GethShotgun</Ammo_Geth>
+		</ammoTypes>
+	</CombatExtended.AmmoSetDef>
+
+	<CombatExtended.AmmoSetDef>
+		<defName>AmmoSet_GethSniper</defName>
+		<label>Geth thermal clip</label>
+		<ammoTypes>
+			<Ammo_Geth>Bullet_GethSniper</Ammo_Geth>
+		</ammoTypes>
+	</CombatExtended.AmmoSetDef>
+
+	<CombatExtended.AmmoSetDef>
+		<defName>AmmoSet_GethSpitfire</defName>
+		<label>Geth thermal clip</label>
+		<ammoTypes>
+			<Ammo_Geth>Bullet_GethSpitfire</Ammo_Geth>
+		</ammoTypes>
+	</CombatExtended.AmmoSetDef>
+
+  <!-- ==================== AmmoSet ========================== -->
+
+  <CombatExtended.AmmoSetDef>
+    <defName>AmmoSet_ML77CE</defName>
+    <label>ML-77 Missile</label>
+    <ammoTypes>
+      <Ammo_ML77CE>Bullet_ML77CE</Ammo_ML77CE>
+    </ammoTypes>
+  </CombatExtended.AmmoSetDef>
+
+	<!-- ==================== Ammo ========================== -->
+
+	<ThingDef Class="CombatExtended.AmmoDef" ParentName="SpacerSmallAmmoBase">
+		<defName>Ammo_Geth</defName>
+		<label>Geth thermal clip</label>
+		<description>Thermal clips designed to fit all geth weapons.</description>
+		<statBases>
+			<Mass>0.01</Mass>
+			<Bulk>0.01</Bulk>
+			<MarketValue>0.24</MarketValue>
+		</statBases>
+		<tradeTags>
+                                                <li>CE_AutoEnableTrade_Sellable</li>
+			<li>CE_AutoEnableCrafting_FabricationBench</li>
+			<!-- Plasma ammo can't be handloaded, and the containment cell must be precision-machined anyway -->
+		</tradeTags>
+		<thingCategories>
+			<li>AmmoPlasmaCellRifle</li>
+		</thingCategories>
+		<graphicData>
+			<texPath>ThirdParty/MassEffect/Heatsink</texPath>
+			<graphicClass>Graphic_StackCount</graphicClass>
```

Comment:
Why not use the custom CE graphic class? One thermal clip can fit multiple shots inside and afaik the graphic class was initially created to be used in cases like this.

Reply:
I'm not sure what's the use of the custom CE GC in this case, using stack count creates the same result.

Reply:
Graphic_StackCount only uses the _a image for a single item in the stack and since a single thermal clip contains multiple "charges" using the _b image when technically there isn't even enough "charges" for filling one thermal clip doesn't make a whole lot of sense.

StackCountRanged can allow you to keep using the first image for a specific amount of items so it doesn't look like you're using the entire clip per shot and it would help with general consistency since [plasma cells](https://github.com/CombatExtended-Continued/CombatExtended/blob/Development/Defs/Ammo/Advanced/PlasmaCellPistol.xml) use the custom graphic and are the closest thing to the thermal clips rn.

If you need a more in-depth explanation the [source file](https://github.com/CombatExtended-Continued/CombatExtended/blob/Development/Source/CombatExtended/CombatExtended/Graphic_StackCountRanged.cs) for the graphic has it.

Reply:
Ok